pyhms/demes/initialize.py

```diff
@@ -1,67 +1,88 @@
+from typing import Any
+
 from pyhms.config import (
     BaseLevelConfig,
     CMALevelConfig,
     DELevelConfig,
     EALevelConfig,
-    LHSLevelConfig,
     LocalOptimizationConfig,
+    RandomLevelConfig,
     SHADELevelConfig,
-    SobolLevelConfig,
 )
+from pyhms.utils.parameter_calculation import get_default_mutation_std
 from structlog.typing import FilteringBoundLogger
 
 from ..core.individual import Individual
 from .abstract_deme import AbstractDeme
 from .cma_deme import CMADeme
 from .de_deme import DEDeme
 from .ea_deme import EADeme
-from .lhs_deme import LHSDeme
 from .local_deme import LocalDeme
+from .random_deme import RandomDeme
 from .shade_deme import SHADEDeme
-from .sobol_deme import SobolDeme
 
 
 def init_root(config: BaseLevelConfig, logger: FilteringBoundLogger) -> AbstractDeme:
-    return init_from_config(config, "root", 0, 0, None, logger)
+    return init_from_config(config, "root", 0, 0, None, None, logger)
+
+
+def prepare_initializer(
+    config: BaseLevelConfig,
+    target_level: int,
+    sprout_seed: Individual | None,
+    injected_population: list[Individual] | None,
+) -> None:
+    context: dict[str, Any] = {}
```

Comment:
I really don't like the use of context dictionaries in this code. It makes the code harder to understand and maintain.

Lack of Clarity: When you use a context dictionary, it's not immediately clear what fields are expected or required. There's no explicit contract or structure, so you have to dig through the code or documentation (if it even exists) to figure out what keys need to be provided. This ambiguity can easily lead to mistakes, like forgetting a key or passing the wrong data type, which might not be caught until runtime.

Increased Risk of Errors: Since the dictionary keys are often just strings, it's easy to make a typo or use an incorrect key, leading to subtle bugs that can be tricky to diagnose. For example, if the code expects a key like "seed_genome" but you accidentally pass "seedGenome", the error might not surface until much later, potentially in a different part of the program.